Can't login to a self-hosted non-ssl site

### Expected behavior
Self-hosted site login to a non-ssl site works on all Android versions.

### Actual behavior
When I try to login to a self-hosted site which doesn't use SSL (https), the login fails with the following error `CLEARTEXT communication to XYZ not permitted by network security policy`. 


Note: 
It seems cleartext communication is disabled by default on API 28+. This issue started appearing since we recently incremented the `targetSdkVersion` from v26 to v28.

### Steps to reproduce the behavior
1. Use a device with Android 9+
2. Try to login to a self hosted site which doesn't use ssl (https) [eg. create a site on poopy.life]
3. Notice the login fails


##### Tested on [device], Android [version], WPAndroid [version]
Pixel 3 and Emulator API 29

This issue is present in v12.9 and needs to be fixed before full-release. cc @loremattei @jtreanor 

cc @maxme

I confirm this is an issue in v12.9, got the same error on a non-ssl self hosted site:
```
2019-07-22 10:06:39.782 4542-4542/org.wordpress.android.beta E/WordPress-API: Volley error on
    com.android.volley.NoConnectionError: java.net.UnknownServiceException: CLEARTEXT communication to do.wpmt.co not permitted by network security policy
        at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:158)
        at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:114)
     Caused by: java.net.UnknownServiceException: CLEARTEXT communication to do.wpmt.co not permitted by network security policy
```

--- a/WordPress/src/main/res/xml/network_security_config.xml
+++ b/WordPress/src/main/res/xml/network_security_config.xml
@@ -1,5 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
 <network-security-config>
+    <!-- Clear text traffic is required for communication with self hosted site without SSL -->
+    <!-- The default configuration for apps targeting Android 7.0 (API level 24) to Android 8.1 (API level 27) is as follows:  -->
+    <!-- https://developer.android.com/training/articles/security-config#CleartextTrafficPermitted -->
+    <base-config cleartextTrafficPermitted="true">
+        <trust-anchors>
+            <certificates src="system"/>
+        </trust-anchors>
+    </base-config>
     <debug-overrides>
         <trust-anchors>
             <!-- Trust user added certificate authorities while debuggable only -->

